src/DssSpell.t.sol

```diff
@@ -543,4 +541,166 @@ contract DssSpellTest is DssSpellTestBase {
+        flash.vatDaiFlashLoan(address(this), 1 * MILLION * RAD, "");
+        assertEq(vat.sin(address(flash)), 0);
+        assertEq(vat.dai(address(flash)), 0);
+        flash.accrue();
+        assertEq(vat.dai(address(flash)), 0);
+        assertEq(vat.dai(address(vow)), vowDai);
+    }
+
+    // Test to ensure 1 Dai flash mint fails after spell
+    function testFlashFailsAfterSpell() public {
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        // Give ourselves tokens for repayment in the callbacks
+        _giveTokens(address(dai), 1_000 * WAD);
+
+        FlashAbstract flash = FlashAbstract(addr.addr("MCD_FLASH_LEGACY"));
+
+        // Fail Here
+        vm.expectRevert("DssFlash/ceiling-exceeded");
+        flash.flashLoan(address(this), address(dai), 1 * WAD, "");
+    }
+
+    function onFlashLoan(
+        address initiator,
+        address token,
+        uint256 amount,
+        uint256 fee,
+        bytes calldata
+    ) external returns (bytes32) {
+        assertEq(initiator, address(this));
+        assertEq(token, address(dai));
+        assertEq(amount, 1 * MILLION * WAD);
+        assertEq(fee, 0);
+
+        dai.approve(msg.sender, 1_000_000 * WAD);
+
+        return keccak256("ERC3156FlashBorrower.onFlashLoan");
+    }
+
+    function onVatDaiFlashLoan(
+        address initiator,
+        uint256 amount,
+        uint256 fee,
+        bytes calldata
+    ) external returns (bytes32) {
+        assertEq(initiator, address(this));
+        assertEq(amount, 1 * MILLION * RAD);
+        assertEq(fee, 0);
+
+        vat.move(address(this), msg.sender, 1_000_000 * RAD);
+
+        return keccak256("VatDaiFlashBorrower.onVatDaiFlashLoan");
+    }
+
+    function testAaveV2D3MRemoved() public {
+
+        DirectDepositLike aaveD3M = DirectDepositLike(addr.addr("MCD_JOIN_DIRECT_AAVEV2_DAI"));
+        ClipAbstract aaveD3MClip = ClipAbstract(addr.addr("MCD_CLIP_DIRECT_AAVEV2_DAI"));
+
+        // Norevert
+        chainLog.getAddress("MCD_JOIN_DIRECT_AAVEV2_DAI");
+        chainLog.getAddress("MCD_CLIP_DIRECT_AAVEV2_DAI");
+        chainLog.getAddress("MCD_CLIP_CALC_DIRECT_AAVEV2_DAI");
+
+        (string memory name, string memory symbol, uint256 class, uint256 dec, address gem, address pip, address join, address xlip) = reg.info("DIRECT-AAVEV2-DAI");
+        assertEq(name, "Aave interest bearing DAI");
+        assertEq(join, address(aaveD3M));
+        assertEq(xlip, address(aaveD3MClip));
+
+        assertEq(vat.wards(address(aaveD3M)), 1);
+        assertEq(vat.wards(address(aaveD3MClip)), 1);
+        assertEq(aaveD3M.wards(pauseProxy), 1);
+        assertEq(aaveD3MClip.wards(pauseProxy), 1);
+
+        assertEq(aaveD3M.live(), 1);
+        assertEq(aaveD3MClip.stopped(), 3);
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
```

Comment:
can we add a check before casting that the balance of the D3M is 0?

Reply:
98e1e6de41177183e94299fbb744750cf7cbb678